File: cs/Shared/InputReader.cs
namespace aoc24;

public static class InputReader
{
    public const string NewLine = "\r\n";

    private static string? rootDirectory = null;

    public static string ReadProblemInput(string id) => File.ReadAllText($"{GetRootDirectory()}/input/{id}.txt");

    private static string GetRootDirectory()
    {
        if (rootDirectory is not null)
            return rootDirectory;

        var directory = new DirectoryInfo(Environment.CurrentDirectory);

        while (directory!.Name != "cs")
            directory = directory.Parent;

        rootDirectory = directory.Parent!.ToString();
        return rootDirectory;
    }
}

File: go/shared/inputReader.go
package shared

import (
	"io"
	"os"
)

var NewLine = "\r\n"

func ReadProblemInput(problem string) (string, error) {
	file, err := os.Open("../../input/" + problem + ".txt")
	if err != nil {
		return "", err
	}
	defer file.Close()

	content, err := io.ReadAll(file)
	if err != nil {
		return "", err
	}
	return string(content), nil
}
